### PR TITLE
Improve markdown rendering of commands help for shell completion

### PR DESCRIPTION
**Pull request by zigarn**
**What this PR does / why we need it**: Allow to have a readable doc on the onine Helm doc website

**Special notes for your reviewer**: Cf. helm/helm-www#1100

**If applicable**:
- [x] this PR contains documentation
- [ ] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Review by marckhouzam — APPROVED** [review]
LGTM

This will make the helm.sh docs look much better for the completion commands.  Thanks @zigarn!

As the patch is officially bigger than XS, it will need another maintainer to approve.

I suggest we pick this into the 3.6 RC so that the docs look better.

**Review by bacongobbler — APPROVED** [review]
Also LGTM. I'll leave it to the release engineer in charge to determine if this should go into 3.6 or not. If not we'll merge this and it'll be available in Helm 3.7. Thanks!

**Comment by marckhouzam** [comment]
I've asked and the policy is that only regressions can be added to an RC.  So this will need to wait until the 3.7 release.
With some luck, the RC will have no problems and it will be the final 3.6 release, so it makes sense not to add anything unless it is absolutely critical.
I will merge into main.

**Pull request merged**
